collection/collection/build.gradle

```diff
@@ -64,16 +66,32 @@ kotlin {
                 dependsOn commonMain
             }
 
-            configure([linuxX64Main, macosX64Main, macosArm64Main]) {
-                dependsOn nativeMain
-            }
-
             nativeTest {
                 dependsOn commonTest
             }
 
-            configure([linuxX64Test, macosX64Test, macosArm64Test]) {
-                dependsOn nativeTest
+            macMain {
+                dependsOn(nativeMain)
+            }
+            linuxMain {
+                dependsOn(nativeMain)
+            }
+
+            targets.all { target ->
+                if (target.platformType == KotlinPlatformType.native) {
```

Comment:
To me, the previous way is more readable and has less code.

```
darwinMain.dependsOn nativeMain

configure([macosX64Main, macosArm64Main]) {
    dependsOn darwinMain
}

linuxMain.dependsOn nativeMain

configure([linuxX64Main]) {
    dependsOn linuxMain
}
```

Reply:
so what i hated about the old one is that, i was trying to change available platforms to test things and i have to update sourceSets for them. ideally, all of this goup will be in the KMP abstraction of AndroidX. Until then, this feels safer?